ipld/write.go

```diff
@@ -55,3 +48,12 @@ func ExtractODSShares(eds *rsmt2d.ExtendedDataSquare) [][]byte {
 	}
 	return origShares
 }
+
+// batchSize calculates the amount of nodes that are generated from block of 'squareSizes'
+// to be batched in one write.
+func batchSize(squareSize int) int {
+	// (squareSize*squareSize) - all the shares
+	// (squareSize*2-1) - amount of nodes in a generated binary tree
+	// *squareSize*2 - multiplier for the amount of trees, both over rows and cols
+	return squareSize*squareSize + (squareSize*2-1)*squareSize*2
```

Comment:
Ideally, this would be exactly the number of all nodes of all trees of one block of total width `squareSize`, right?

Reply:
Yes, I hope so, but I could make a mistake here.

Reply:
ah, I see. Then this is correct!

Reply:
maybe add a comment in the code as well why we count the leaves twice?

Reply:
Good

Reply:
> Yes, but do you remember that in IPLD implementation we have one more link per each leave.


Can you see a reason we would ever remove that property and have the hashes in the leaf nodes directly? I can't think of a good reason at least..

Reply:
> Can you see a reason we would ever remove that property and have the hashes in the leaf nodes directly?

I think we should remove this property. This only adds additional roundtrips on the network + more block data size amplification. I remember you did this for a reason and I remember that I had a solution for that reason in my mind, but not in detail right now. Also, IIRC, I mentioned this in the storage optimization issue.

Reply:
Hmm, then it will be tricky to remember to change this here as well. If there is an issue for this (the one you mention about optimizing storage) it should probably link to this PR and the above method even IMO.

Reply:
Just added more info regarding the tree, will also add a link to storage optimization with cross-linking